### PR TITLE
Practice manager - Looking for feedback

### PR DESCRIPTION
**Pull request by bretto36**
Hi All

We use the Xero API but also the Xero Practice Manager API, we've recently had to move to OAUTH2 and am hoping to use the calcinai package so i don't have to reinvent the wheel. 

Is it possible to get some feedback on this so i don't go too far down the rabbit hole doing it the wrong way.

### CODE REVIEW

**Comment by bretto36** [comment]
@calcinai Hi! Thank you for the great package. Hoping i could get some of your time to review a Pull Request and tell me if i'm on the right track to add Practice Management support

**Comment by calcinai** [comment]
@bretto36 looking great! Are you using this in your application currently?

**Comment by bretto36** [comment]
@calcinai  - Yes currently using it in a staging environment. About to go live with it. It's not the full API, but it does handle the features we needed to bring across now that the switch to OAUTH2.0 is on. 

These are some of the ways we are using it. 

`$query = $this->getApi($account)->load(\XeroPHP\Models\PracticeManager\Invoice::class)
                      ->setParameter('detailed', 'true')
                      ->setParameter('from', $dateFrom)
                      ->setParameter('to', $dateTo);

        /** @var \XeroPHP\Models\PracticeManager\Invoice[] $invoices */
        $invoices = $query->execute();`

`/** @var \XeroPHP\Models\PracticeManager\CustomField[] $customFields */
        $customFields = $this->getApi($account)->load(\XeroPHP\Models\PracticeManager\CustomField::class)->execute();`

`$query = $this->getApi($account)->load(\XeroPHP\Models\PracticeManager\Client::class);

        if (!$force) {
            $query->setParameter('modifiedsince', $now->subDays(7)->format('Y-m-d\TH:i:s'));
        }

        /** @var \XeroPHP\Models\PracticeManager\Client[] $clients */
        $clients = $query->execute();

foreach ($clients as $client) {
    foreach ($client->getCustomFieldValues() as $customFieldValue) {
        // DO something here
    }
}`

**Comment by bretto36** [comment]
Just thought i'd update here. Have been using this in production for over 18 months now. Working well.

File: src/XeroPHP/Traits/PracticeManager/CustomFieldValueTrait.php [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+     * @throws \XeroPHP\Remote\Exception\UnauthorizedException
+     */
+    public function getCustomFieldValues()
+    {
+        /**
+         * @var \XeroPHP\Remote\Model
+         */
+        if ($this->hasGUID() === false) {
+            throw new Exception(
+                'Attachments are only available to objects that exist remotely.'
```
[the remark this answers is not in the document]
Comment (bretto36) [reply src/XeroPHP/Traits/PracticeManager/CustomFieldValueTrait.php]: @calcinai  Have updated the message. Thanks for spotting it

**Comment by calcinai** [comment]
Nice work @bretto36

**Comment by bretto36** [comment]
@calcinai Thanks mate. Appreciate the work you've done on the package so far. Come in very handy for me personally. Especially when Xero retired the oauth1a.

**Pull request merged**